extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/AbstractMongoClientProducer.java

```diff
@@ -217,11 +219,24 @@ private MongoClientSettings createMongoConfiguration(MongoClientConfig config) {
         }
         // add pojo codec provider with automatic capabilities
         // it always needs to be the last codec provided
-        CodecProvider pojoCodecProvider = PojoCodecProvider.builder()
+        PojoCodecProvider.Builder pojoCodecProviderBuilder = PojoCodecProvider.builder()
                 .automatic(true)
-                .conventions(Conventions.DEFAULT_CONVENTIONS)
-                .build();
-        providers.add(pojoCodecProvider);
+                .conventions(Conventions.DEFAULT_CONVENTIONS);
+        // register bson discriminators
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        if (classLoader == null) {
```

Comment:
I don't think this can happen really... And if it does it's doubtful that `AbstractMongoClientProducer.class.getClassLoader()` will save you (certainly not dev mode that is :))

Reply:
I copied this pattern from the Smallrye Fault Tolerance extension, and I just check almost all extensions directly used `Thread.currentThread().getContextClassLoader()` so it should be safe to use it.

Reply:
Yeah, I can't see a scenario where `Thread.currentThread().getContextClassLoader()` will return null. In any case this is very minor so it's not worth putting too much thought into it